google/cloud/storage/internal/grpc_split_write_object_data.cc

```diff
@@ -21,20 +21,7 @@ namespace cloud {
 namespace storage_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
-SplitObjectWriteData<std::string>::SplitObjectWriteData(
-    absl::string_view buffer)
-    : buffers_({buffer}),
-      total_size_(storage::internal::TotalBytes(buffers_)) {}
-
-SplitObjectWriteData<std::string>::SplitObjectWriteData(
-    google::cloud::storage::internal::ConstBufferSequence buffers)
-    : buffers_(std::move(buffers)),
-      total_size_(storage::internal::TotalBytes(buffers_)) {}
-
-bool SplitObjectWriteData<std::string>::Done() const {
-  return offset_ >= total_size_;
-}
-
+template <>
 std::string SplitObjectWriteData<std::string>::Next() {
   auto constexpr kMax = static_cast<std::size_t>(
       google::storage::v2::ServiceConstants::MAX_WRITE_CHUNK_BYTES);
```

Comment:
PS: In the `std::string` case, it _might_ behoove you to precompute the final size and reserve space up front (assuming that copying dominates).

Reply:
Good idea, will do.

Reply:
Given your clarification regarding `kMax`, perhaps a simple `result.reserve(kMax)` would be best.